lib/src/working_copy.rs

```diff
@@ -85,7 +87,15 @@ impl FileState {
         FileState {
             file_type: FileType::Conflict { id },
             mtime: mtime_from_metadata(metadata),
-            size,
+            size
+        }
+    }
+
+    fn for_gitsubmodule(id: CommitId, metadata: &Metadata) -> Self {
+        FileState {
+            file_type: FileType::GitSubmodule { id},
+            mtime: mtime_from_metadata(metadata),
+            size: metadata.len(),
```

Comment:
For submodules, the whole `FileState` is just a placeholder to indicate that there is a submodule that we should ignore, so how about using 0 for both mtime and and size? I don't think they'll be used for anything. That would also let us remove the `metadata` argument.